src/py42/services/watchlists.py

```diff
@@ -0,0 +1,150 @@
+import sys
+
+from py42.exceptions import Py42BadRequestError
+from py42.exceptions import Py42Error
+from py42.exceptions import Py42InvalidWatchlistType
+from py42.exceptions import Py42NotFoundError
+from py42.exceptions import Py42WatchlistNotFound
+from py42.exceptions import Py42WatchlistOrUserNotFound
+from py42.services import BaseService
+from py42.services.util import get_all_pages
+
+
+class WatchlistsService(BaseService):
+
+    _uri_prefix = "/v1/watchlists"
+
+    def __init__(self, connection):
+        super().__init__(connection)
+        self._watchlist_type_id_map = {}
+
+        if "pytest" not in sys.modules:
+            # initiate mapping between types and IDs
+            watchlists = (self.get_page(page_size=100)).data["watchlists"]
```

Comment:
Rather than making this request in init (which means every time anyone initializes the sdk, this call is going to be made), we should move this into the property and called only the first time it's actually accessed: 

```
@property
def watchlist_type_id_map(self):
    if not self._watchlist_type_id_map:
        self._watchlist_type_id_map = self.get_page(page_size=100)).data["watchlists"]
    return self._watchlist_type_id_map
```

Then for testing just set `._watchlist_type_id_map` on the class prior to running the tests to avoid making any actual calls.